### ISSUE TITLE
Simplifying proxy management 

### ISSUE THREAD

**Issue by atsareg** [DIRACGrid/DIRAC#4105]
While reviewing the current proxy management code in preparation of the new OAuth based authentication and related stuff, we identified several functionalities that are either not used or can be considered obsoleted. If this code is removed, the new functionality implementation will become simpler. The following functionalities are to be considered for removal:

First, proxy renewal mechanism using external MyProxy servers. This was introduced at the time when the DIRAC ProxyManager was a new service and users used to use MyProxy services for the same purpose. To ensure a smooth transition, the mechanism for the user proxy renewal by the MyProxyRenewalAgent was introduced. To my knowledge, this mechanism is used nowhere now. Instead, users are supposed to upload long living proxies directly to the ProxyManager. With the use of new mechanisms for proxy provisioning (Proxy providers), the use of long proxies will be reduced. The renewal mechanism will be replaced by provisioing new relatively short living proxies. So, I suggest to remove this mechanism and the related code from DIRAC. If any DIRAC installation uses this functionality or administrators see this to be useful, please, speak up.

Second, we have a mechanism that is limiting a client with a given proxy to retrieve some other's proxy by a predefined number of times, I mean proxy retrieval tokens. This is used now to limit the number of times a pilot can retrieve user proxies. In practice, this number is limited by the number of jobs that pilot can retrieve in Filling mode. On the other hand, as a security measure, this mechanism is rather weak - if a pilot proxy is stolen, it does not matter so much how many user proxies can be retrieved with malicious intentions. For many years of functioning of various DIRAC services, we never suffered from this kind of attacks. Now, with the increasing use of various robot certificates, this kind of protection is becoming redundant and complicating unnecessarily the system. So, the suggestion is to drop it to simplify the code. 

Please, comment.

**Comment by chaen** [comment]
TLDR: I am totally up for simplification, so in general, a BIG YES for the principle :-)

A bit more material now.

` With the use of new mechanisms for proxy provisioning (Proxy providers), the use of long proxies will be reduced.` Possibly for some VOs, but it will clearly not disappear, so that should stay in place.  But removing the `MyProxy` part could probably go away.

`For many years of functioning of various DIRAC services, we never suffered from this kind of attacks` Please.... :-) 

About the functionality itself, I still find this protection useful, and looking at the code, it really is minimal (one update query to decrease the counter, and one extra condition in the query for retrieving the proxy), so I would leave it.

About the actual realisation of all that, it sounds reasonable to postpone these changes to v7r1 at least, given that there are so many pending things already.

**Comment by atsareg** [comment]
OK for MyProxy. As for the tokens, they are also used in the evaluation of the number of pilots to be submitted because the number of jobs per pilots can not be more than the number of tokens as each job spends at least one token. Dropping tokens will allow to drop this logic as well making the whole thing simpler.

**Comment by chaen** [comment]
I think you can remove the logic in the pilot submission without removing the token counters. It then becomes an operational parameter to adjust. It's not really different than a timeout or something like this, you just have to adjust it to your conditions.

**Comment by fstagni** [comment]
Token counter in SiteDirector has been removed long time ago. PR 6451 removes MyProxy support and solves this issue.
